src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -839,11 +841,14 @@ public int Length
             }
         }
 
-        // The mask used when shuffling a single int into Vector128/256.
+        // The shuffle control mask used when shuffling a single int into Vector128/256.
         // On little endian machines, the lower 8 bits of int belong in the first byte, next lower 8 in the second and so on.
+        // (And on big endian, upper 8 bits in the first byte, next upper 8 bits in the second...)
         // We place the bytes that contain the bits to its respective byte so that we can mask out only the relevant bits later.
-        private static readonly Vector128<byte> s_lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
-        private static readonly Vector128<byte> s_upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
+        private static readonly Vector128<byte> s_lowerShuffleMask_CopyToBoolArray =
+            BitConverter.IsLittleEndian ? Vector128.Create(0, 0x01010101_01010101).AsByte() : Vector128.Create(0x03030303_03030303, 0x02020202_02020202).AsByte();
+        private static readonly Vector128<byte> s_upperShuffleMask_CopyToBoolArray =
```

Comment:
With the work done in https://github.com/dotnet/runtime/pull/36267 and #35857, is it still beneficial for these to be cached in static readonly fields?

Reply:
My guess is that we don't need these anymore - if I remember correctly those PRs make the JIT recognise those Create calls and store the data in memory during codegen, so the code can load the data from memory - so I expect it to result in about the same, or perhaps better code. It's been a while since I've rebased the branch so I could make the change while I'm rebasing and see how it turns out.